plugin.video.vrt.nu/resources/lib/kodiwrapper.py

```diff
@@ -635,49 +623,53 @@ def refresh_caches(self, cache_file=None):
-        ''' Invalidate a existing cache file '''
-        self.delete_file(self._cache_path + path)
-
     def invalidate_caches(self, expr=None):
         ''' Invalidate multiple cache files '''
         import fnmatch
         _, files = self.listdir(self._cache_path)
         if expr:
             files = fnmatch.filter(files, expr)
-        for f in files:
-            self.delete_file(self._cache_path + f)
+        for filename in files:
+            self.delete_file(self._cache_path + filename)
+        self.delete_file(self._cache_path + 'favorites.json')
         self.delete_file(self._cache_path + 'oneoff.json')
 
+    @staticmethod
+    def input_down():
+        ''' Move the cursor down '''
+        xbmc.executeJSONRPC('{"jsonrpc": "2.0", "method": "Input.Down", "id": 1}')
+
     def container_refresh(self):
         ''' Refresh the current container '''
-        self.log_notice('Execute: Container.Refresh', 'Debug')
+        self.log('Execute: Container.Refresh', 'Debug')
         xbmc.executebuiltin('Container.Refresh')
 
-    def container_update(self, url=None, path='', replace=False):
-        ''' Update the current container '''
-        if url is None:
-            url = self._url
-        self.log_notice('Execute: Container.Update(%s%s%s)' % (url, path, ',replace' if replace else ''), 'Debug')
-        xbmc.executebuiltin('Container.Update(%s%s%s)' % (url, path, ',replace' if replace else ''))
-
     def end_of_directory(self):
         ''' Close a virtual directory, required to avoid a waiting Kodi '''
         xbmcplugin.endOfDirectory(handle=self._handle, succeeded=False, updateListing=False, cacheToDisc=False)
 
     def log_access(self, url, query_string=None, log_level='Verbose'):
         ''' Log addon access '''
-        if log_levels.get(log_level, 0) <= self._max_log_level:
+        if LOG_LEVELS.get(log_level, 0) <= self._max_log_level:
             message = '[%s] Access: %s' % (self._addon_id, url + ('?' + query_string if query_string else ''))
             xbmc.log(msg=from_unicode(message), level=xbmc.LOGNOTICE)
 
-    def log_notice(self, message, log_level='Info'):
+    def log(self, message, log_level='Info', **kwargs):
         ''' Log info messages to Kodi '''
-        if log_levels.get(log_level, 0) <= self._max_log_level:
-            message = '[%s] %s' % (self._addon_id, message)
-            xbmc.log(msg=from_unicode(message), level=xbmc.LOGNOTICE)
+        cur_log_level = LOG_LEVELS.get(log_level, 0)
+        if not self._debug_logging and 1 < cur_log_level <= self._max_log_level:
+            # If Debug Logging is not enabled, Kodi filters everything up to NOTICE out
+            log_level = 'Info'
+        if kwargs:
+            import string
+            message = string.Formatter().vformat(message, (), SafeDict(**kwargs))
+        message = '[{addon}] {message}'.format(addon=self._addon_id, message=message)
+        xbmc.log(msg=from_unicode(message), level=XBMC_LOG_LEVELS.get(log_level))
```

Comment:
Logging to the Kodi log is only allowed with the `DEBUG` level. See https://kodi.wiki/view/Add-on_rules:

> addons should use the xbmc.log() method to write to the logfile and use the debug logging level only.

Reply:
This is the case. But we make an exception for testing/development.

Reply:
But in that case, you do write to the Kodi log with other levels?

Reply:
This is not resolved yet

> This is the case. But we make an exception for testing/development.

Testing and development should be done before an add-on is submitted. Beside that, I'm very curious who the "we" is in that case. As "we" (in meaning of TeamKodi) haven't made any exception I could think of. So please follow our add-on rules and change what others have requested.

Thanks in advance

Reply:
@DaVukovic When I said *"We make an exception"* I meant that from the settings.xml we (the add-on developers) allow the user to change the log level so that we can have a detailed add-on logging without a detailed Kodi logging. This helps in troubleshooting add-on specific issues.

Reply:
@basrieter Incorrect. By default nothing is logged (this depends on `max_log_level` in *settings.xml*), but when `debug_logging` is `True`, we force `xbmc.LOGDEBUG` for everything and we force the maximum logging in the add-on.

Reply:
So then I can't follow your code. If the `debug_logging` is false, the log level is set to `Info`, which logs in the kodi.log with level `Notice`? Which is against the rules?

Reply:
Well, there does seem to be a problem, but it is not related to the fact we check for debug_logging, it is related to the fact we don't filter out based on max_log_level.

The whole point of this change is NOT to log with LOGNOTICE by default.